execution_engine/src/core/engine_state/balance.rs

```diff
@@ -1,11 +1,43 @@
-use casper_types::{URef, U512};
+use casper_types::{Key, URef, U512};
 
-use crate::shared::newtypes::Blake2bHash;
+use crate::{
+    shared::{newtypes::Blake2bHash, stored_value::StoredValue},
+    storage::trie::merkle_proof::TrieMerkleProof,
+};
 
 #[derive(Debug)]
 pub enum BalanceResult {
     RootNotFound,
-    Success(U512),
+    Success {
+        motes: U512,
+        main_purse_proof: Box<TrieMerkleProof<Key, StoredValue>>,
```

Comment:
this should be `purse_proof` instead of `main_purse_proof`

Reply:
Fixed in https://github.com/CasperLabs/casper-node/pull/438/commits/49f9f4b5b9ec8ed59730b0ba65f4992b66597632